src/Build/Evaluation/PropertyTrackingEvaluatorDataWrapper.cs

```diff
@@ -323,11 +336,13 @@ private void TrackPropertyReassignment(P? predecessor, P property, string? locat
                 (_settings == 0 && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10)))
             {
                 var args = new PropertyReassignmentEventArgs(
-                        property.Name,
-                        oldValue,
-                        newValue,
-                        location,
-                        message: null)
+                    property.Name,
+                    oldValue,
+                    newValue,
+                    location?.File,
+                    location?.Line ?? 0,
+                    location?.Column ?? 0,
+                    message: ResourceUtilities.GetResourceString("PropertyReassignment"))
```

Comment:
why do we need this? I think we can leave the message null.

Even if we needed this, we could call GetResourceString once and cache it, since it never changes, and it's pretty expensive.